src/testsuite/arith128_test_f128.c

```diff
@@ -0,0 +1,4254 @@
+const vui64_t vf128_nmax  = CONST_VINT64_DW(0xfffeffffffffffff, 0xffffffffffffffff);
+
+const vui64_t vf128_min   = CONST_VINT64_DW(0x0001000000000000, 0x0000000000000000);
+const vui64_t vf128_nmin  = CONST_VINT64_DW(0x8001000000000000, 0x0000000000000000);
+
+const vui64_t vf128_sub   = CONST_VINT64_DW(0x0000ffffffffffff, 0xffffffffffffffff);
+const vui64_t vf128_nsub  = CONST_VINT64_DW(0x8000ffffffffffff, 0xffffffffffffffff);
+
+const vui64_t vf128_inf   = CONST_VINT64_DW(0x7fff000000000000, 0);
+const vui64_t vf128_ninf  = CONST_VINT64_DW(0xffff000000000000, 0);
+
+const vui64_t vf128_nan   = CONST_VINT64_DW(0x7fff800000000000, 0);
+const vui64_t vf128_nnan  = CONST_VINT64_DW(0xffff800000000000, 0);
+
+const vui64_t vf128_snan  = CONST_VINT64_DW(0x7fff400000000000, 0);
+const vui64_t vf128_nsnan = CONST_VINT64_DW(0xffff000080000000, 0);
+
+const vui32_t vf128_true = CONST_VINT32_W(0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff);
+const vui32_t vf128_false = CONST_VINT32_W(0, 0, 0, 0);
+
+int
+test_isinf_signf128 (void)
+{
+  const __binary128 f128_zero = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x0000000000000000, 0));
+  const __binary128 f128_nzero = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x8000000000000000, 0));
+
+  const __binary128 f128_one = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x3fff000000000000, 0));
+  const __binary128 f128_none = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0xbfff000000000000, 0));
+
+  const __binary128 f128_max = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x7ffeffffffffffff, 0xffffffffffffffff));
+  const __binary128 f128_nmax = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0xfffeffffffffffff, 0xffffffffffffffff));
+
+  const __binary128 f128_min = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x0001000000000000, 0x0000000000000000));
+  const __binary128 f128_nmin = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x8001000000000000, 0x0000000000000000));
+
+  const __binary128 f128_sub = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x0000ffffffffffff, 0xffffffffffffffff));
+  const __binary128 f128_nsub = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x8000ffffffffffff, 0xffffffffffffffff));
+
+  const __binary128 f128_inf = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x7fff000000000000, 0));
+  const __binary128 f128_ninf = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0xffff000000000000, 0));
+
+  const __binary128 f128_nan = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x7fff800000000000, 0));
+  const __binary128 f128_nnan = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0xffff800000000000, 0));
+
+  const __binary128 f128_snan = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0x7fff400000000000, 0));
+  const __binary128 f128_nsnan = vec_xfer_vui64t_2_bin128 (
+      CONST_VINT128_DW(0xffff000080000000, 0));
+
+  __float128 x;
+  long tests_count = 0;
+  int test, expt;
```

Comment:
`expt` is not really needed, as it is always just set and immediately passed as a parameter.
_global comment_